machine.go

```diff
@@ -1067,3 +1067,88 @@ func (m *Machine) CreateSnapshot(ctx context.Context, memFilePath, snapshotPath
 	m.logger.Debug("snapshot created successfully")
 	return nil
 }
+
+// CreateBalloon creates a balloon device if one does not exist
+func (m *Machine) CreateBalloon(ctx context.Context, amountMb int64, deflateOnOom bool, statsPollingIntervals int64, opts ...PutBalloonOpt) error {
+	balloon := models.Balloon{
+		AmountMb:              &amountMb,
+		DeflateOnOom:          &deflateOnOom,
+		StatsPollingIntervals: statsPollingIntervals,
+	}
+	_, err := m.client.PutBalloon(ctx, &balloon, opts...)
+
+	if err == nil {
+		m.logger.Printf("Created balloon device with %v MB successful", amountMb)
+	} else {
+		m.logger.Errorf("Create balloon device failed : %s", err)
+	}
+	return err
+}
+
+// GetBalloonConfig gets the current balloon device configuration.
+func (m *Machine) GetBalloonConfig(ctx context.Context) (models.Balloon, error) {
+	var balloonConfig models.Balloon
+	resp, err := m.client.DescribeBalloonConfig(ctx)
+	if err != nil {
+		m.logger.Errorf("Getting balloonConfig: %s", err)
+		return balloonConfig, err
+	}
+
+	balloonConfig = *resp.Payload
+	m.logger.Printf("GetBalloonConfig successful")
+	return balloonConfig, err
+}
+
+// UpdateBalloon will update an existing balloon device, before or after machine startup
+func (m *Machine) UpdateBalloon(ctx context.Context, amountMb int64, opts ...PatchBalloonOpt) error {
+	ballonUpdate := models.BalloonUpdate{
+		AmountMb: &amountMb,
+	}
+	respNoContent, err := m.client.PatchBalloon(ctx, &ballonUpdate, opts...)
+	if err == nil {
+		m.logger.Printf("Update balloon device with %v MB successful: %s", amountMb, respNoContent.Error())
+	} else {
+		m.logger.Errorf("Update balloon device failed : %s", err)
+	}
```

Comment:
nit: I think this might be slightly more idiomatic if it was if err != nil {....} removing the need for the else.

Reply:
Fixed it.